conjure-lib/src/main/java/com/palantir/conjure/java/lib/internal/ConjureClientEndpoint.java

```diff
@@ -0,0 +1,35 @@
+/*
+ * (c) Copyright 2021 Palantir Technologies Inc. All rights reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.palantir.conjure.java.lib.internal;
+
+import java.lang.annotation.ElementType;
+import java.lang.annotation.Retention;
+import java.lang.annotation.RetentionPolicy;
+import java.lang.annotation.Target;
+
+/**
+ * Signifies that the given method corresponds to a endpoint invoked by a conjure client.
+ */
+@Retention(RetentionPolicy.CLASS)
+@Target(ElementType.METHOD)
+public @interface ConjureClientEndpoint {
+    String method();
+
+    String path();
+
+    String name();
```

Comment:
service-endpoint and method-path should both be unique identifiers, I'm not sure we need both

Reply:
No, we don't need both necessarily, but I was experimenting with switching to service-endpoint so added that.  It turns out using service-endpoint isn't as easy as hoped - the code that generates the method code only has an EndpointDefinition available to it, not a  ServiceDefinition, so we don't have access to the service name. I don't want to go modifying all the endpoint generation code just to pass the service name through.

An alternative would be to use the Service annotation in combination with the Endpoint annotation - getting the service name from one and the endpoint from the other. I know you didn't like having the other annotation, but it may be more clear to readers that way.

Regardless, I think having the name in the Endpoint annotation doesn't do any harm, and makes it easier to recognize when reading.

Reply:
For what it's worth, changing code-generating code is pretty low lift, so I wouldn't be opposed to the refactoring, if it's useful.

Reply:
+1, I'd prefer modifying the code-gen code rather than sacrificing legibility of the code it produces.

Reply:
What I've found useful in dialogue-annotations-processor is instantiating the *Generator classes per ServiceDefinition, instad of one per JVM and then passing ServiceDefinition to #generate method: it simplified a whole bunch of state threading to not have to pass stuff as method args through a million different codepaths. Maybe that could work here to simplify stuff?

Reply:
I really don't feel comfortable re-writing the generators for this small change. I'll do another commit passing the ServiceDefinition through the couple places needed

Reply:
Carter and I chatted offline and I think came to the conclusion that the http method-path combo which is there now is fine because it's the most concise form of annotation without introducing in new (and fragile) conventions for specifying a package+service+endpoint unique name as a string. So I'm not going to push anything new unless he comes back and says I missed something.